### PR TITLE
Add --op-return flag to `ord wallet send`

### LINKED ISSUE #3007: Feature request: `ord wallet send` has the possibility of adding `OP_RETURN` data
The use-case of this could be:
- on-chain visibility of some important details when transferring the inscription

### PR DESCRIPTION
**Pull request by EricGrill**
## Summary

Fixes #3007

This PR adds the ability to include OP_RETURN data when sending inscriptions, sats, runes, or bitcoin amounts via `ord wallet send`.

### Changes
- Add `--op-return` flag to the send command that accepts up to 80 bytes of data
- Validate that OP_RETURN data does not exceed the 80-byte Bitcoin standard limit
- Append OP_RETURN output (with zero value) to the transaction

### Use Cases
- Adding on-chain metadata when transferring inscriptions (e.g., price paid, transaction notes)
- Including payment memos or references visible on block explorers
- Embedding arbitrary data alongside transfers

### Example Usage
```bash
# Send an inscription with metadata
ord wallet send --fee-rate 5 --op-return "Sold for 0.5 BTC" bc1q... <inscription_id>

# Send bitcoin with a memo
ord wallet send --fee-rate 5 --op-return "Invoice #12345" bc1q... 1btc
```

## Test plan
- [x] Add test for sending inscription with OP_RETURN data
- [x] Add test for sending BTC amount with OP_RETURN data  
- [x] Add test for error when OP_RETURN data exceeds 80 bytes
- [x] Add test for OP_RETURN with exactly 80 bytes (boundary condition)
- [x] Add test for dry-run mode with OP_RETURN

## Notes
- Reference implementation from PR #3067 was consulted
- OP_RETURN outputs are appended after the main transaction is built, ensuring no interference with ordinal-aware transaction construction
- The OP_RETURN output has zero value and uses standard `OP_RETURN <data>` script format

:robot: Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: tests/wallet/send.rs [hunk 1]
```diff
@@ -1311,3 +1311,200 @@ fn error_messages_use_spaced_runes() {
...
+
+  create_wallet(&core, &ord);
+
+  let txid = core.mine_blocks(1)[0].txdata[0].compute_txid();
+
+  // Create data that exceeds 80 bytes
+  let op_return_data = "a".repeat(81);
+
+  CommandBuilder::new(format!(
+    "wallet send --fee-rate 1 --op-return '{}' bc1qw508d6qejxtdg4y5r3zarvary0c5xw7kv8f3t4 {txid}:0:0",
```
Comment (letetrautre) [inline tests/wallet/send.rs]: bc1qz02ya56ut7u2jgakdflmslqev6zzmnr36wfhwu

**Review by letetrautre — APPROVED** [review]

File: src/subcommand/wallet/send.rs
Comment (letetrautre) [inline src/subcommand/wallet/send.rs]: 0xe79638f89130988cbd27c51064186bef46797865

File: src/subcommand/wallet/send.rs [hunk 2]
```diff
@@ -38,7 +44,16 @@ impl Send {
...
+      ensure!(
+        data.len() <= 80,
+        "OP_RETURN data exceeds maximum size of 80 bytes ({} bytes provided)",
+        data.len()
+      );
+    }
+
+    let mut unsigned_transaction = match self.asset {
       Outgoing::Amount(amount) => {
         wallet.create_unsigned_send_amount_transaction(address, amount, self.fee_rate)?
```
Comment (letetrautre) [inline src/subcommand/wallet/send.rs]: 0xe79638f89130988cbd27c51064186bef46797865
